Add PlanPriceDisplay Component

robertf4 opened the pull request:
#### Changes proposed in this Pull Request:
* This adds a PlanPriceDisplay component that will be used in https://github.com/Automattic/jetpack/pull/14018.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
See the P2 here for the design of this PR: p1HpG7-7MK-p2 (specifically the part titled "WP Admin Desktop")
See the P2 here for the overall MT: p1HpG7-7ET-p2

#### Testing instructions:
This cannot be tested on its own here. Use #14018 to test.

#### Proposed changelog entry for your changes:
* No changelog needed

robertf4 committed:
[not verified] Add products endpoint

robertf4 committed:
[not verified] Add products endpoint

robertf4 committed:
[not verified] Add QueryProducts data component and corresponding Redux and rest API calls

robertf4 committed:
[not verified] [not verified] Add error code

robertf4 committed:
[not verified] [not verified] Add error code

robertf4 committed:
[not verified] Changed benefits to products

robertf4 committed:
[not verified] Update _inc/client/state/site/reducer.js

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

robertf4 committed:
[not verified] Create PlanRadioButton

robertf4 committed:
[not verified] Create PlanPriceDisplay

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -6,6 +6,35 @@ import { translate as __ } from 'i18n-calypso';
+
+	return (
+		<div className="single-product-backup__plan-price-display-container">
+			<div className="discounted-price__container">
```
Since it's the "root" of the CSS block, why are we calling this `discounted-price__container` and not just `discounted-price`?

Also, do we really need two separate CSS blocks for `discounted-price` and `plans-price`? Couldn't we just have a `plan-price` block that would accept a modifier like `is-discounted`? The modifier would change the color of the text and add the slash (most likely we could add the slash as a pseudoelement like `::before`).

robertf4 replied:
I like this idea, but I'd like to do it in another PR after this is merged to make testing easier.

delawski replied:
Sounds good 👍

robertf4 replied:
Created https://github.com/Automattic/jetpack/issues/14051

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -6,6 +6,35 @@ import { translate as __ } from 'i18n-calypso';
+export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+	const perYearPriceRange = `${ currencySymbol }${ backupPlanPrices.jetpack_backup_daily.yearly }-${ backupPlanPrices.jetpack_backup_realtime.yearly } /year`;
+
+	const full_daily_backup_yearly_cost = backupPlanPrices.jetpack_backup_daily.monthly * 12;
```
Shouldn't this and the next line be in camelCase?

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -6,6 +6,35 @@ import { translate as __ } from 'i18n-calypso';
 import './single-product-backup.scss';
 
+export function PlanPriceDisplay( { backupPlanPrices, currencySymbol } ) {
+	const perYearPriceRange = `${ currencySymbol }${ backupPlanPrices.jetpack_backup_daily.yearly }-${ backupPlanPrices.jetpack_backup_realtime.yearly } /year`;
```
This line is very long and har to read. What do you think of assigning the daily yearly and realtime yearly prices to some clearly-named constants first and then use them here? We could do the same with the monthly prices used below.

Also, there's a hardcoded "year" string inside. We need to make it available for translation.

jeherve replied:
In addition to translating `/year`, I think we'd also need to be able to localize the whole string. In some countries the order of the words may be "/year - price - currency" for example.

robertf4 replied:
Agreed for both issues.

jeherve replied:
Do you think you could create issues, or one issue to keep track of the things that need to be fixed before launch, like the above?

robertf4 replied:
> Do you think you could run `yarn build` in the environment where you commit, to avoid these errors and benefit from the things our pre-commit hook brings us, such as linting and prettyfying on commit?

I do this. For this case, because I split out each component into its own PR, the components were unused which gives an error that prevented me from committing. I only committed when this was the only error I got.

robertf4 replied:
I will do that, but in this case the issue is fixed in this PR.

delawski commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -6,6 +6,35 @@ import { translate as __ } from 'i18n-calypso';
+	const full_realtime_backup_yearly_cost = backupPlanPrices.jetpack_backup_realtime.monthly * 12;
+
+	return (
+		<div className="single-product-backup__plan-price-display-container">
```
While this class name is long, it doesn't seem to be too descriptive. Those plan prices live in the single product backup header, so the class name like:

```scss
.single-product-backup__header-price
```
would be shorter and describe the element clearer in my opinion. What do you think?

robertf4 replied:
Agreed.

delawski reviewed (changes requested):
@robertf4 Thanks for your work on this. I left some comments in the code.

I know you've got a lot on your plate right now, so if you need any support in addressing the feedback, feel free to reach out to me.

robertf4 committed:
[not verified] Switch to camel case

robertf4 committed:
[not verified] I18N perYearPriceRange

robertf4 committed:
[not verified] Factor out variables for readability

robertf4 committed:
[not verified] Rename single-product-backup__plan-price-display-container to single-product-backup__header-price

delawski reviewed (dismissed):
Thank you for updating the code.

I'm approving it as is, also keeping in mind that we'll still have to refactor the price blocks in one of the next PRs.

jeherve commented:
> pre-commit hook was skipped for one or more commits

Do you think you could run `yarn build` in the environment where you commit, to avoid these errors and benefit from the things our pre-commit hook brings us, such  as linting and prettyfying on commit?

robertf4 commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -6,6 +6,51 @@ import { translate as __ } from 'i18n-calypso';
+	const realtimeBackupYearlyPrice = backupPlanPrices.jetpack_backup_realtime.yearly;
+	const realtimeBackupMonthlyPrice = backupPlanPrices.jetpack_backup_realtime.monthly;
+
+	const fullDailyBackupYearlyCost = dailyBackupMonthlyPrice * 12;
```
For this case I am going off of the logic that @enejb described to me, which they are also using in Calypso. The calculation here is getting the price we have discounted from for the yearly subscription, which is monthly * 12.

robertf4 replied:
This is the same logic used in Calypso according to @enejb, and the point of the discounted is to show how buying a yearly subscription is cheaper than paying by month.

gravityrail replied:
@robertf4 is right, closing this comment.

robertf4 commented:
> Do you think you could create issues, or one issue to keep track of the things that need to be fixed before launch, like the above?

I certainly can, but in this case the I18N issue is fixed in this PR.

robertf4 commented:
> Do you think you could run `yarn build` in the environment where you commit, to avoid these errors and benefit from the things our pre-commit hook brings us, such as linting and prettyfying on commit?

I do this. For this case, because I split out each component into its own PR, the components were unused which gives an error that prevented me from committing. I only committed when this was the only error I got.

gravityrail reviewed (dismissed):
This is good to go IMO. A few follow-up PRs but nothing that should block this.

robertf4 committed:
Add SingleProductBackupBody Component (#14027)

* Add products endpoint

* [not verified] Add error code

* Update _inc/lib/class.core-rest-api-endpoints.php

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

* Update _inc/lib/class.core-rest-api-endpoints.php

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

* Add QueryProducts data component and corresponding Redux and rest API calls

* Update _inc/client/state/site/reducer.js

Co-Authored-By: Marin Atanasov <8436925+tyxla@users.noreply.github.com>

* [not verified] Add SingleProductBackupBody component

* [not verified] Rename plans-section__body to single_product_backup__body

* [not verified] Give class name to h4 tag

* [not verified] Remove single_product_backup__body

* [not verified] Check for upgradeLinks and upgradeTitles before using them

* [not verified] Switch function style

* [not verified] Add key to PlanRadioButton

* Remove duplicate definition of getProducts

* Allow TOS agreement before Jetpack is fully active so we track… (#14041)

* Allow TOS agreement before Jetpack is fully active so we track the connection flow

* Update packages/terms-of-service/src/class-terms-of-service.php

Co-Authored-By: Derek Smart <smart@automattic.com>

* Also update the action that is being called in jetpack

We renamed the action lets also rename it in Jetpack.

* [not verified] Revert "Also update the action that is being called in jetpack"

This reverts commit e9c0f21d51151900e218e8253832ab1a5245575e.


Co-authored-by: Brandon Kraft <public@brandonkraft.com>
Co-authored-by: Enej Bajgoric <enej.bajgoric@gmail.com>

* Removed Jetpack references in the IXR client. (#14046)

This change allows the package to be used outside of Jetpack, relying on the methods provided by the Manager, plus using the WP_Error class instead of the Jetpack_Error wrapper.

* Full Sync: Don't allow more than one request to enqueue (#14039)

* Spell checking CI integration (#13992)

* Adds spell checking and fixes files.

This implements a package to spellcheck files, and
does a pass to ensure everything is green.

* [not verified] Implements Travis check for spelling

Adds Yarn script and Travis CI for spell checking.

* [not verified] Fixes feedback.

Updates docs to prevent spelling exceptions for certain files.

* Export SingleProductBackupBody class

* [not verified] No need to export single product backup body

gravityrail committed:
Merge branch 'master' into add/plan-price-display-component

gravityrail committed:
Merge branch 'add/plan-price-display-component' of github.com:Automattic/jetpack into add/plan-price-display-component

gravityrail committed:
Ignore unused var for now

gravityrail reviewed (dismissed):
Good to go

gravityrail committed:
Ignore unused var

jeherve commented:
We can probably close this PR in favor of #14056.
